### PR TITLE
Custom vision 2.0

### PR DESCRIPTION
**Pull request by areddish**
## Description
Regenerate client for v2.0 Swagger
Update tests for new functionality
Update tests to be able to record session when needed

Public submission from private PR#454
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
- Please reference the private repo PR corresponding to this 
- Please pull down latest changes from upstream and run code generation again

**Comment by areddish** [comment]
Private repo PR is https://github.com/Azure/azure-sdk-for-net-pr/pull/832
Swagger PR is https://github.com/Azure/azure-rest-api-specs/pull/3026

Pulled upstream, rebased off it and re-ran code gen, only the metadata changed.

File: src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Prediction/Microsoft.Azure.CognitiveServices.Vision.CustomVision.Prediction.csproj [hunk 1]
```diff
@@ -6,14 +6,15 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.CognitiveServices.Vision.CustomVision.Prediction</PackageId>
     <Description>This client library provides access to the Microsoft Cognitive Services CustomVision Prediction APIs.</Description>
-    <VersionPrefix>0.9.0-preview</VersionPrefix>
+    <Version>0.10.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Prediction/Microsoft.Azure.CognitiveServices.Vision.CustomVision.Prediction.csproj]: There is a diff here with the private repo PR.
Also, PackageReleaseNotes indicate this is not a preview release but the Version does, please verify the intent here

File: src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training/Microsoft.Azure.CognitiveServices.Vision.CustomVision.Training.csproj [hunk 2]
```diff
@@ -6,15 +6,15 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.CognitiveServices.Vision.CustomVision.Training</PackageId>
     <Description>This client library provides access to the Microsoft Cognitive Services CustomVision Training APIs.</Description>
-    <VersionPrefix>1.0.2-preview</VersionPrefix>
+    <Version>0.10.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training/Microsoft.Azure.CognitiveServices.Vision.CustomVision.Training.csproj]: Don't think you can go back versions, also differs from the private repo entry, please check the versioning here

File: src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training.Tests/BaseTests.cs [hunk 3]
```diff
@@ -1,30 +1,226 @@
...
-        protected ITrainingApi GetTrainingApiClient(DelegatingHandler handler)
+        protected ITrainingApi GetTrainingApiClient()
         {
-            ITrainingApi client = new TrainingApi(handlers: handler)
+            ITrainingApi client = new TrainingApi(handlers: HttpMockServer.CreateInstance())
             {
-                ApiKey = TrainingKey
+                ApiKey = TrainingKey,
+#if RECORD_MODE
+                //BaseUri = new Uri("https://deviris2.azure-api.net/V2.0/Training") //new Uri("http://localhost:8080/api_v2.0/training")
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training.Tests/BaseTests.cs]: nit: remove commented code

**Review by dsgouda — CHANGES_REQUESTED** [review]
Versioning inconsistent with the private repo PR, please verify

File: src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Prediction/Microsoft.Azure.CognitiveServices.Vision.CustomVision.Prediction.csproj (hunk 1 above)
Comment (areddish) [reply src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Prediction/Microsoft.Azure.CognitiveServices.Vision.CustomVision.Prediction.csproj]: Yes, I was asked to modify the version and update the preview outside of the private pr. I can send you the info so you can validate.

File: src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training.Tests/BaseTests.cs [hunk 4]
```diff
@@ -46,10 +46,6 @@ protected ITrainingApi GetTrainingApiClient()
             ITrainingApi client = new TrainingApi(handlers: HttpMockServer.CreateInstance())
             {
                 ApiKey = TrainingKey,
-#if RECORD_MODE
-                //BaseUri = new Uri("https://deviris2.azure-api.net/V2.0/Training") //new Uri("http://localhost:8080/api_v2.0/training")
-                BaseUri = new Uri("https://cognitivescusppe.azure-api.net/customvision/v2.0/Training")
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training.Tests/BaseTests.cs]: Do we still need the `BaseUri` in RECORD_MODE?

**Review by dsgouda — COMMENTED** [review]
LGTM apart from the question

File: src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training.Tests/BaseTests.cs (hunk 4 above)
Comment (areddish) [reply src/SDKs/CognitiveServices/dataPlane/Vision/CustomVision/Training.Tests/BaseTests.cs]: It depends on how it's recorded, I'm not using it for Prediction so I'm keeping it consistent between Prediction and Training.

**Review by dsgouda — APPROVED** [review]
Will merge on CIs passing

**Pull request merged**